docs/build/guides/conventions/analyzing-smart-contract-cost.mdx

```diff
@@ -0,0 +1,449 @@
+---
+title: Analyzing smart contract cost
+hide_table_of_contents: false
+---
+
+Several factors influence how quickly and efficiently your smart contracts execute on the Stellar network. This guide will help you understand these factors and provide tips on how to write cost-effective contracts.
+
+# **How to optimize smart contract cost**:
+
+Complex contracts with numerous conditions, loops, and computations require more processing power on Stellar. This can lead to higher gas costs (transaction fees) and slower execution times.
+
+## 1. **Efficient loop and storage calls usage**
+
+A contract that requires multiple loops and conditions to execute will cost more than a simple contract that executes a single operation.
+
+#### Not Optimal Contract ❎
+
+```rust
+#![no_std]
+use soroban_sdk::{contract, contractimpl, log, vec,Map, Env};
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    // Function to update values in storage inefficiently
+    pub fn update_values(env: Env, values: Vec<u32>) {
+        for &value in values.iter() {
+            let current_count = env.storage().persistent().get("total_count");
+            env.storage().persistent().set("total_count", &(current_count + value));
+        }
+    }
+}
+
+```
+
+:::danger
+
+_Problem_ : Each iteration of the loop performs a separate read and write operation to update total_count in storage.
+
+_Inefficient_: This results in multiple expensive storage operations (read and write) within the loop, increasing gas costs significantly as the array size (values.len()) grows.
+
+:::
+
+#### Optimal Contract ✅
+
+```rust
+#![no_std]
+use soroban_sdk::{contract, contractimpl, log, vec, Env, Map};
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    // Function to update values in storage efficiently
+    pub fn update_values(env: Env, values: Vec<u32>) {
+        let mut total_count = env.storage().persistent().get("total_count");
+
+        for &value in values.iter() {
+            total_count += value;
+        }
+
+        env.storage().persistent().set("total_count", &total_count);
+    }
+}
+
+```
```

Comment:
This is a great example since it highlights how easy it is to program the "typical" way for a dev vs. the optimal way in the context of a smart contract. Another really useful example here I'd like to see would be subcontract calls, perhaps just using a token transfer, to highlight another case of accumulation vs. step-by-step transfers.